plugin/ai_helper/tool/learnpath.php

```diff
@@ -25,10 +27,21 @@
         $wordsCount = (int) $_REQUEST['words_count'];
         $courseCode = (string) $_REQUEST['course_code'];
         $sessionId = (int) $_REQUEST['session_id'];
+        $addTests = ('true' === $_REQUEST['add_tests']);
+        $nQ = ($addTests ? (int) $_REQUEST['nro_questions'] : 0);
 
         $messageGetItems = 'Generate the table of contents of a course in "%s" in %d or less chapters on the topic of "%s" in a list separated with comma, without chapter number';
         $prompt = sprintf($messageGetItems, $courseLanguage, $chaptersCount, $topic);
         $resultText = $plugin->openAiGetCompletionText($prompt, 'learnpath');
+
+        if (isset($resultText['error']) && true === $resultText['error']) {
+            echo json_encode([
+                'success' => false,
+                'text' => $resultText['message'],
+            ]);
+            exit;
+        }
+
```

Comment:
@christianbeeznest Parece que algo está mal con el orden.
@FraGoTe sugirió añadir `krsort($items)` justo antes del `foreach($items as $item)` unas líneas a bajo, pero también falla para mi. También intenté usar `ksort` en vez de `krsort`, pero tampoco es perfecto.
Como los tests se generan en base a los pasos, estos siempre caen bien (justo detrás del documento sobre el cual hacen preguntas), pero el problema está en los documentos.
Intenté, en un curso en Inglés, el query "History of Uganda" con 5 páginas y 300 palabras por página (y 2 preguntas por test). Luego intenté "History of Botswana" y me dió un orden ilógico también.
Se puede evidenciar en que hablan del periodo pre-colonial *después* de hablar de la historia actual, por ejemplo.
Supongo que el problema proviene de OpenAI mismo, que no devuelve los elementos en orden.

Reply:
Hola @ywarnier

He probado poniendo "History of Peru" varias veces y  se guarda con el orden de lista que envia el openAi, he probado el texto en el chat GPT y me sale el mismo resultado. Quizas siendo mas especifico en el pedido por ejemplo "History of Peru in chronological order" saldria mejor el resultado.

Reply:
![Capture d’écran du 2023-02-15 14-23-45](https://user-images.githubusercontent.com/609839/219039412-8dce961b-30a2-47b8-936e-4321e871cd20.png)
Según la base de datos, los pre-requisitos están bien, pero el previous_item_id, next_item_id y display_order están mal, dejando a la interfaz el lujo de ordenar los pasos como quiera.

Reply:
![Capture d’écran du 2023-02-15 14-31-16](https://user-images.githubusercontent.com/609839/219041054-485ef59a-3233-4d0f-985e-02ce12a09efe.png)
Haciendo el cambio manualmente, todo funciona bien.

Reply:
Hola @ywarnier 
Cierto, por alguna razon el display order no estaba guardando , veo que necesitaba el previous item, acabo de agregarlo en el siguiente commit, y parece estar todo bien ahora.

Gracias.